src/hook.c

```diff
@@ -757,3 +761,46 @@ MH_STATUS WINAPI MH_ApplyQueued(VOID)
         LeaveSpinLock();
     }
 }
+
+//-------------------------------------------------------------------------
+MH_STATUS WINAPI MH_CreateHookApi(LPCSTR pszTarget, LPVOID pDetour, LPVOID *ppOriginal)
+{
+    HMODULE hModule;
+    LPVOID  pTarget;
+    SIZE_T  copySize;
+    LPCSTR  psz = pszTarget;
+    LPCSTR  pszProcName = NULL;
+    CHAR    szModuleName[MAX_PATH];
+
+    if (pszTarget == NULL || pDetour == NULL)
+        return MH_ERROR_INVALID_ARGUMENTS;
+
+    // search the last '.' (replacement for strrchr())
+    while (*psz)
+        if (*psz++ == '.')
+            pszProcName = psz;
```

Comment:
> Or do you mean something else?
That depends from compiler...

examples: 
- GCC compiler: http://i.imgur.com/hvQjVnt.png?1
- Borland C++ Builder 6: http://i.imgur.com/oBg4uHQ.png?1
- Embarcadero C++ Builder XE7: http://i.imgur.com/cHnLuQr.png?1